geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegionMap.java

```diff
@@ -2112,9 +2130,16 @@ public void txApplyPut(Operation putOp, Object key, Object nv, boolean didDestro
       callbackEvent.makeSerializedNewValue();
       txHandleWANEvent(owner, callbackEvent, txEntryState);
     }
-    RegionMapCommitPut commitPut = new RegionMapCommitPut(this, owner, callbackEvent, putOp,
-        didDestroy, txId, txEvent, pendingCallbacks, txEntryState);
-    commitPut.put();
+    boolean locked = owner.lockWhenRegionIsInitializing();
```

Comment:
It would be better for this code to be added to RegionMapCommitPut instead of AbstractRegionMap.
In the RegionMapCommitPut is should be done following the pattern in that class of a method that takes a Runnable.
For an example see: org.apache.geode.internal.cache.map.AbstractRegionMapPut.runWhileLockedForCacheModification(Runnable)
I think you would want to do this in: AbstractRegionMapPut because it would be the same for both a CommitPut and the non-commit put